### PR TITLE
Remove Hadoop Configuration dependency in BaseMetastoreTableOperations

### PR DESCRIPTION
**Pull request by jerryshao**
This is for issue #353 , to remove the Configuration dependency in `BaseMetastoreOperations`. 

No new UT added, using existing UTs.

CC @aokolnychyi , please help to review, thanks!

### CODE REVIEW

File: core/src/main/java/org/apache/iceberg/BaseMetastoreTableOperations.java [hunk 1]
```diff
@@ -43,17 +41,15 @@
...
 
   private TableMetadata currentMetadata = null;
   private String currentMetadataLocation = null;
   private boolean shouldRefresh = true;
   private int version = -1;
 
-  protected BaseMetastoreTableOperations(Configuration conf) {
-    this.conf = conf;
-    this.fileIo = new HadoopFileIO(conf);
+  protected BaseMetastoreTableOperations(FileIO fileIo) {
```
Comment (aokolnychyi) [inline core/src/main/java/org/apache/iceberg/BaseMetastoreTableOperations.java]: As an alternative, we can consider providing an abstract method that subclasses should implement. That way, we have more flexibility over initializing `FileIO`. Calling parent constructors must always be the first call in child constructors, which means we will have problems when initializing `FileIO` requires calling another method. 

@jerryshao  what do you think?

Comment (jerryshao) [reply core/src/main/java/org/apache/iceberg/BaseMetastoreTableOperations.java]: OK, let me update the code.

Comment (jerryshao) [reply core/src/main/java/org/apache/iceberg/BaseMetastoreTableOperations.java]: @aokolnychyi please help to review again, thanks!

**Comment by rdblue** [comment]
+1

Thanks for fixing this, @jerryshao!

**Pull request merged**